volatility3/framework/plugins/linux/pagecache.py

```diff
@@ -458,11 +454,34 @@ def write_inode_content_to_file(
                             inode_size,
                             page_idx,
                         )
-                    f.seek(current_fp)
-                    f.write(page_bytes)
+                    buffer.seek(current_fp)
+                    buffer.write(page_bytes)
+
+            except IOError as e:
+                vollog.error(f"Unable to read inode: {e!s}")
+                return None
+
+            buffer.seek(0)
+            return buffer.read()
+
+    def write_inode_content_to_file(
```

Comment:
By not parameterizing `open` this method is only usable by this plugin.  We try to encourage methods that could be reusable to be made class or static methods, and parameterized with everything they need (such as an open method).  That also changes the API in a non-backwards compatible way, so if were to allow it (unlikely) it would require a major version bump of the plugin.  Better to just parameterize with the same parameters as the original version and then no version bumps required.

Reply:
Thanks for your review! I changed `write_inode_content_to_file` back to the original parameters. I did make `write_inode_content_to_file` a `classmethod` (which shouldn't impact callers of the method), for easier access to `read_inode_content`, but I can change it back to a `staticmethod` if you wish.

I also did not do a (minor) version bump, but I could set `InodePages._version` to `(1, 0, 2)` if you want.

Reply:
Nope, class method is fine, it just needs to have the same function signature.  You should bump the minor version number though, so that tools that require the new function can ensure they're using a version that includes it.  `1.0.2` would be a patch bump, not a minor bump.  It would need to be `1.1.0`